### ISSUE TITLE
Add Foldable.surround

### ISSUE THREAD

**Issue by mlms13** [reazen/relude#56]
```reason
List.surround((module String.Semigroup), "-", []); // "-"
List.surround((module String.Semigroup), "a", ["h"]); // "aha"
List.surround((module String.Semigroup), "*", ["1", "2", "3"]); // "*1*2*3*"
```

This is a lot like `Foldable.intercalate` except the "separator" is also added to the beginning and end. This also means that the inner type only needs to implement Semigroup (not Monoid) because the separator is used as the initial value.

**Comment by andywhite37** [comment]
I added `surround` and `surroundMap` which existed in bs-abstract `Functions` modules.  I'm not sure if that's the same function you're talking about here.

**Comment by andywhite37** [comment]
This is released in 0.12.0.  Re-open if the `FoldableSemigroupExtensions` version of this function is not what you were talking about.
